### PR TITLE
Apps : Fix Hidden Secrets Not Saving in Dot Velocity Secrets

### LINKED ISSUE #25855: Apps : Fix Hidden Secrets Not Saving in Dot Velocity Secrets
### Problem Statement
In the app "Dot Velocity Secrets" the hidden secrets are not getting saved. Users are inputting the secret but it is not being saved and therefore inaccessible in the app.

### Steps to Reproduce
1. Go to apps > Dot Velocity Secrets app
2. Create a new hidden value
4. Click save
5. Observe that the secret is not saved

### Acceptance Criteria
We need to make sure that we hide the value in the UI, but we actually send the real value to the backend. 

### External Links
https://github.com/dotCMS/core/assets/751424/4bd91927-411d-4f05-ba26-b9ee9dc8d735

### Assumptions & Initiation Needs
We assume that the problem is related to the backend code not properly saving the hidden values.

### Quality Assurance Notes & Workarounds
We can test the functionality by manually inputting hidden values in the UI and then verifying that they are saved in the backend. As a workaround, users can manually re-enter the secret each time they need to access it.

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Fix hidden secrets not saving in dot velocity secrets

### Checklist
- [x] Tests

### <samp>🤖 Generated by Copilot at d452604</samp>

This pull request improves the key-value component and its subcomponents by refactoring the code, adding test coverage, and enhancing security and testability. It changes the files `dot-key-value-ng.component.ts` and `dot-key-value-ng.component.spec.ts` to handle the `password` variable differently, and the files `dot-key-value-table-row.component.html`, `dot-key-value-table-row.component.ts`, and `dot-key-value-table-row.component.spec.ts` to add a password placeholder and a `data-testId` attribute.

## Related Issue
Fixes #25855

# Explanation of Changes

### <samp>🤖 Generated by Copilot at d452604</samp>

*  Remove the logic of setting the hidden value of the variable to asterisks from the key-value component and move it to the key-value table row component, to simplify the code and avoid duplication ([link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-86b4c43a2626753069032d582c7d79ac4d6d3964a46a5c137edb7af3655a4be2L49-L57), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-86b4c43a2626753069032d582c7d79ac4d6d3964a46a5c137edb7af3655a4be2L69-L74), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-2610ac0d211d109fffa59c8ea3894a5575317a2819226334f138de3a324ec719L6-R7), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-2610ac0d211d109fffa59c8ea3894a5575317a2819226334f138de3a324ec719L24-R32))
*  Define a property and a constant for the password placeholder in the key-value table row component and its test, to avoid hard-coding the value and make it easier to change if needed ([link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-12d6f6b26442ea2a7a6a977e8407dcf66b456bb9bad8e41f64ee62eaa8f9f5dfR38), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-e26382376f0631375534fa80290146501db94b8751349124717ac0c0da85aef4R29-R30))
*  Add a data-testId attribute to the table cell that displays the value of the variable in the key-value table row component, to make it easier to query in the test ([link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-2610ac0d211d109fffa59c8ea3894a5575317a2819226334f138de3a324ec719L6-R7))
*  Update the key-value component test and add a new test case for the key-value table row component test, to check that the password placeholder is shown instead of the actual value if the variable is hidden ([link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-62c75db396c499f43b54eddb350886de53ad5aea490effcb15e07199d931884fL28-R28), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-62c75db396c499f43b54eddb350886de53ad5aea490effcb15e07199d931884fL179-R178), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-e26382376f0631375534fa80290146501db94b8751349124717ac0c0da85aef4R264-R271))
*  Remove the unused import of the DotKeyValueUtil class from the key-value component ([link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-86b4c43a2626753069032d582c7d79ac4d6d3964a46a5c137edb7af3655a4be2L7-L8))
*  Remove some unnecessary line breaks in the templates of the key-value component, the key-value table row component, and their test host components ([link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-62c75db396c499f43b54eddb350886de53ad5aea490effcb15e07199d931884fL37-R37), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-2610ac0d211d109fffa59c8ea3894a5575317a2819226334f138de3a324ec719L46-R44), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-2610ac0d211d109fffa59c8ea3894a5575317a2819226334f138de3a324ec719L55-R52), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-2610ac0d211d109fffa59c8ea3894a5575317a2819226334f138de3a324ec719L63-R59), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-2610ac0d211d109fffa59c8ea3894a5575317a2819226334f138de3a324ec719L70-R70), [link](https://github.com/dotCMS/core/pull/26023/files?diff=unified&w=0#diff-e26382376f0631375534fa80290146501db94b8751349124717ac0c0da85aef4L37-R39))

### Screenshots

https://github.com/dotCMS/core/assets/72418962/fcc390a9-959e-4708-8c1b-a4e3e86df341

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/view/components/dot-key-value-ng/dot-key-value-table-row/dot-key-value-table-row.component.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ export class DotKeyValueTableRowComponent implements OnInit, OnChanges {
     @Output() cancel: EventEmitter<number> = new EventEmitter(false);
     @Output() delete: EventEmitter<DotKeyValue> = new EventEmitter(false);
 
+    passwordPlaceholder = '********';
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-key-value-ng/dot-key-value-table-row/dot-key-value-table-row.component.ts:38]: should we set this to `readonly`?

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-key-value-ng/dot-key-value-table-row/dot-key-value-table-row.component.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-key-value-ng/dot-key-value-table-row/dot-key-value-table-row.component.ts:38]: I'll change it. Still in procress tho

File: core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration-detail/dot-apps-configuration-detail-form/dot-apps-configuration-detail-form.component.ts [hunk 2]
```diff
@@ -49,9 +49,8 @@ export class DotAppsConfigurationDetailFormComponent implements OnInit {
             this.emitValues();
         });
 
-        this.emitValues();
-
         setTimeout(() => {
+            this.emitValues();
             this.formContainer.nativeElement.querySelector(`#${this.formFields[0].name}`).focus();
         }, 0);
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration-detail/dot-apps-configuration-detail-form/dot-apps-configuration-detail-form.component.ts:55]: This looks like needs to be in the `ngAfterViewInit`

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration-detail/dot-apps-configuration-detail-form/dot-apps-configuration-detail-form.component.ts [hunk 3]
```diff
@@ -49,9 +49,8 @@ export class DotAppsConfigurationDetailFormComponent implements OnInit {
...
     }
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration-detail/dot-apps-configuration-detail-form/dot-apps-configuration-detail-form.component.ts:56]: This looks like needs to be in the ngAfterViewInit

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by manuelrojas — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**